src/main/java/com/shade/decima/ui/data/viewer/texture/exporter/TextureExporterTGA.java

```diff
@@ -0,0 +1,107 @@
+        header.put((byte) 0); // colourmaptype
+        header.put((byte) (bitsPerPixel == 8 ? 3 : 2)); // datatypecode
+        header.putShort((short) 0); // colourmaporigin
+        header.putShort((short) 0); // colourmaplength
+        header.put((byte) 0); // colourmapdepth
+        header.putShort((short) 0); // x_origin
+        header.putShort((short) 0); // y_origin
+        header.putShort((short) image.getWidth());//width
+        header.putShort((short) image.getHeight());//height
+        header.put((byte) bitsPerPixel);//bitsperpixel
+        header.put((byte) 0);//imagedescriptor
+
+        channel.write(header.position(0));
+
+        ByteBuffer byteBuffer;
+
+        DataBuffer dataBuffer = provider.getImage(0, 0).getData().getDataBuffer();
+
+        if (dataBuffer instanceof DataBufferByte) {
+            if (bitsPerPixel != 8) {
+                throw new IllegalStateException("Expected 8 bits per pixel for Byte data buffer");
+            }
+            byte[] pixelData = ((DataBufferByte) dataBuffer).getData();
+            byteBuffer = ByteBuffer.wrap(pixelData);
+        } else if (dataBuffer instanceof DataBufferUShort) {
+            if (bitsPerPixel != 16) {
+                throw new IllegalStateException("Expected 16 bits per pixel for Short data buffer");
+            }
+            short[] pixelData = ((DataBufferUShort) dataBuffer).getData();
+            byteBuffer = ByteBuffer.allocate(pixelData.length * 2);
+            byteBuffer.asShortBuffer().put(ShortBuffer.wrap(pixelData));
+        } else if (dataBuffer instanceof DataBufferShort) {
+            if (bitsPerPixel != 16) {
+                throw new IllegalStateException("Expected 16 bits per pixel for Short data buffer");
+            }
+            short[] pixelData = ((DataBufferShort) dataBuffer).getData();
+            byteBuffer = ByteBuffer.allocate(pixelData.length * 2);
+            byteBuffer.asShortBuffer().put(ShortBuffer.wrap(pixelData));
+        } else if (dataBuffer instanceof DataBufferInt) {
+            if (bitsPerPixel != 32 && bitsPerPixel != 24) {
+                throw new IllegalStateException("Expected 32/24 bits per pixel for Int data buffer");
+            }
+            int[] pixelData = ((DataBufferInt) dataBuffer).getData();
+            final int bandCount = bitsPerPixel / 8;
+            byteBuffer = ByteBuffer.allocate(pixelData.length * bandCount);
+            int width = image.getWidth();
+            int height = image.getHeight();
+            for (int x = 0; x < width; x++) {
+                for (int y = 0; y < height; y++) {
+                    int val = pixelData[y + height * (width-x-1)];
+                    for (int b = 0; b < bandCount; b++) {
+                        byteBuffer.put((byte) (val >>> (b * 8)));
+                    }
+                }
+            }
+
+        } else {
+            throw new IllegalArgumentException("Not implemented for data buffer type: " + dataBuffer.getClass());
+        }
```

Comment:
This doesn't look right. We need to find a better solution.

Additionally, it's better to validate data before writing something to a file.